addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
+        var text, xPos;
+        // Pick the current text element by filtering on data
+        text = selection.filter(function(data) {
+          return data === d;
+        });
+        // Calculate current horizontal translation from the center
+        xPos = d3.transform(text.attr("transform")).translate[0];
```

Comment:
I will add more comments to explain this line. Essentially I am a) reading the `transform` attribute value of the text element and then, b) parsing it via `d3.transform` method and then c) getting the translateX value stored at "0" index of translate property.